core/templates/dev/head/pages/story-editor-page/editor-tab/story-node-editor.directive.ts

```diff
@@ -95,6 +104,13 @@ angular.module('oppia').directive('storyNodeEditor', [
             $scope.editableOutline = $scope.getOutline();
             $scope.explorationId = $scope.getExplorationId();
             $scope.currentExplorationId = $scope.explorationId;
+            $scope.expIdIsValid = true;
+            if ($scope.currentExplorationId) {
```

Comment:
I'd say allow saving empty exploration IDs here, since if a creator wants to swap exp IDs of 2 chapters, there is no way to do that without creating a dummy exploration as well. 
Also, the validation checks in story-navbar, would prevent the user from saving the exploration anyway, so it wouldn't affect the datastore also, what do you think? If it's fine, @nishantwrp can you make sure that expIdIsValid is true if expId is an empty string as well?

Reply:
@aks681 I think instead of allowing empty explorationId we can create button like `Remove this exploration from this card`. What do you think?

Reply:
I think the "remove exploration" button, as @nishantwrp suggests, would be better. Having the creator type in an empty explorationId seems like a hack to me, and not very intuitive.

@nishantwrp by "card" you mean "node", I think? An exploration corresponds to a chapter/node in a story, and an exploration has multiple "cards".

Reply:
@seanlip I tried adding a remove exploration button, but it turns out there is a lot of backend validation to avoid the same. So we would have change a lot of validations to do that.

I think we can skip the same exploration checking while updating the exploration id for a chapter and can include that in the navbar validation.

Reply:
Yeah, that seems fine, though make sure the user is not able to 'Save Changes' without adding an exploration ID to every chapter.

Reply:
I'm having some trouble understanding what is being talked about here. Could you please clarify in more detail what (a) "the same" in "avoid the same", and (b) "same exploration checking" mean?

Reply:
- `"the same"` - I mean that there is a backend validation to not allow empty exploration id's.
-  `same exploration checking` - I mean that we could add a check in the navbar that disallows more than one chapters having same exploration id's.

Reply:
@seanlip Please reply.

Reply:
Hi @nishantwrp -- sorry for the late reply. This issue is still on my radar but I haven't had time to think about it in more detail. I'll try and get back to you within the next 24 hours!

Reply:
OK, took a look. Let's discuss this in the LaCE meeting. I think that story chapters should allow exploration IDs which are None (but not '') -- it simply means that an exploration ID has not been assigned yet. If this is currently the case, I am not sure why "remove exploration" would lead to a lot of backend validations failing.

For the second part, I think having a navbar check sounds fine. It is a story-level check rather than a chapter-level check, after all.

Reply:
Hi @seanlip ,
There is a problem with this solution too. I'm sorry for not checking that before. Actually there is a backend validation for checking if there are same explorations in the story when we try to save changes. So, even if frontend allows to switch the explorations between the chapters the backend won't allow that when we try to save changes because it processes the both changes seperately. I think we need to build that functionality both in backend and frontend seperately. Any suggestions?